elixir/apps/domain/lib/domain/actors/group/sync.ex

```diff
@@ -47,12 +47,19 @@ defmodule Domain.Actors.Group.Sync do
   end
 
   defp plan_groups_update(groups, provider_identifiers) do
+    identifiers_set = MapSet.new(provider_identifiers)
```

Comment:
Using a MapSet here reduces the runtime of this from O(n^2) to O(n) which might shave a few CPU % off the syncs here.

Reply:
Yeah, it'd be nice if it did.  At this point we may not see much of a change, but at least moving forward it should scale well.